article/app/controllers/ArticleController.scala

```diff
@@ -117,16 +121,27 @@ class ArticleController extends Controller with RendersItemResponse with Logging
 
   def renderLiveBlog(path: String, page: Option[String] = None) =
     Action.async { implicit request =>
-      mapModel(path, blocks = true, page) {// temporarily only ask for blocks too for things we know are new live blogs until until the migration is done and we can always use blocks
-        render(path, _)
+      val range = page.map(ParseBlockId.fromPageParam) match {
+        case Some(Some(id)) => Left(PageWithBlock(id)) // we know the id of a block
+        case Some(None) => Right(NotFound) // page param there but couldn't extract a block id
+        case None => Left(Canonical) // no page param
```

Comment:
Some(Some(..)) and Some(None) sounds weird to me. Would it make sense to create classes that encapsulate the different states?